src/Uno.UI.Composition/Composition/SkiaGeometrySource2D.skia.cs

```diff
@@ -8,16 +8,15 @@ namespace Microsoft.UI.Composition
 {
 	internal class SkiaGeometrySource2D : IGeometrySource2D
 	{
-		public SkiaGeometrySource2D()
-		{
-			Geometry = new SKPath();
-		}
-
 		public SkiaGeometrySource2D(SKPath source)
 		{
 			Geometry = source ?? throw new ArgumentNullException(nameof(source));
 		}
 
-		public SKPath Geometry { get; set; }
+		/// <remarks>
+		/// DO NOT MODIFY THIS SKPath. CREATE A NEW SkiaGeometrySource2D INSTEAD.
+		/// This can lead to nasty invalidation bugs where the SKPath changes without notifying anyone.
+		/// </remarks>
+		public SKPath Geometry { get; }
```

Comment:
Should we refactor this class to keep this path private ?

Reply:
@ramezgerges ? If possible I would expose needed APIs on this `SkiaGeometrySource2D` and keep that `Geometry` private rather than expose something that is risky.

Reply:
I'm open to the idea (but maybe in a separate PR), but I'm not really sure how you want this to go? What would we expose instead of the SKPath? Also, this doesn't fix the problem of someone with a reference to the SKPath (before constructing the SkiaGeometrySource2D) modifying it. I thought about cloning the SKPath in the constructor, but I'm not sure about the performance implications.

Reply:
Let's resolve for now, can be handled in separate PR. The class itself is internal anyway, so it should not be a big problem